src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1606,5 +1608,10 @@ public void ForkChoiceUpdated(Hash256? finalizedBlockHash, Hash256? safeBlockHas
                 _metadataDb.Set(MetadataDbKeys.SafeBlockHash, Rlp.Encode(SafeHash!).Bytes);
             }
         }
+        
+        public Block[] GetInvalidBlocks()
+        {
+            return _invalidBlocks.ToArray().Select(x => x.Value).ToArray();
```

Comment:
Double ToArray, totally not needed. You can add `IEnumerable` implementation to cache if needed.

Reply:
Internally the `LruCache` stores entries as `Dictionary<TKey, LinkedListNode<LruCacheItem>>`, what we want is a `IEnumerable<KeyValuePair<TKey, TValue>>`. How can I deal with the linked list other than doing what `toArray` already does?

Reply:
I'm not sure how to remove either `ToArray`